packages/flutter_goldens/test/skia_client_test.dart

```diff
@@ -0,0 +1,632 @@
+        environment: <String, String>{
+          'FLUTTER_ROOT': _kFlutterRoot,
+        },
+        operatingSystem: 'macos'
+    );
+
+    skiaClient = SkiaGoldClient(
+      workDirectory,
+      fs: fs,
+      process: process,
+      platform: platform,
+      httpClient: fakeHttpClient,
+    );
+
+    traceID = skiaClient.getTraceID('flutter.golden.1');
+    expect(
+      traceID,
+      equals('9968695b9ae78cdb77cbb2be621ca2d6'),
+    );
+  });
+
+  test('throws for error state from imgtestAdd', () {
+    final File goldenFile = fs.file('/workDirectory/temp/golden_file_test.png')
+      ..createSync(recursive: true);
+    platform = FakePlatform(
+        environment: <String, String>{
+          'FLUTTER_ROOT': _kFlutterRoot,
+          'GOLDCTL' : 'goldctl',
+        },
+        operatingSystem: 'macos'
+    );
+
+    skiaClient = SkiaGoldClient(
+      workDirectory,
+      fs: fs,
+      process: process,
+      platform: platform,
+      httpClient: fakeHttpClient,
+    );
+
+    const RunInvocation goldctlInvocation = RunInvocation(
+      <String>[
+        'goldctl',
+        'imgtest', 'add',
+        '--work-dir', '/workDirectory/temp',
+        '--test-name', 'golden_file_test',
+        '--png-file', '/workDirectory/temp/golden_file_test.png',
+        '--passfail',
+      ],
+      null,
+    );
+    process.processResults[goldctlInvocation] = ProcessResult(123, 1, 'Expected failure', 'Expected failure');
+    process.fallbackProcessResult = ProcessResult(123, 1, 'Fallback failure', 'Fallback failure');
+
+    expect(
+      skiaClient.imgtestAdd('golden_file_test', goldenFile),
+      throwsA(
+        isA<SkiaException>().having((SkiaException error) => error.message,
+          'message',
+          contains('result-state.json'),
+        ),
+      ),
+    );
+  });
+
+  test('throws for error state from tryjobAdd', () {
+    final File goldenFile = fs.file('/workDirectory/temp/golden_file_test.png')
+      ..createSync(recursive: true);
+    platform = FakePlatform(
+        environment: <String, String>{
+          'FLUTTER_ROOT': _kFlutterRoot,
+          'GOLDCTL' : 'goldctl',
+        },
+        operatingSystem: 'macos'
+    );
+
+    skiaClient = SkiaGoldClient(
+      workDirectory,
+      fs: fs,
+      process: process,
+      platform: platform,
+      httpClient: fakeHttpClient,
+    );
+
+    const RunInvocation goldctlInvocation = RunInvocation(
+      <String>[
+        'goldctl',
+        'imgtest', 'add',
+        '--work-dir', '/workDirectory/temp',
+        '--test-name', 'golden_file_test',
+        '--png-file', '/workDirectory/temp/golden_file_test.png',
+        '--passfail',
+      ],
+      null,
+    );
+    process.processResults[goldctlInvocation] = ProcessResult(123, 1, 'Expected failure', 'Expected failure');
+    process.fallbackProcessResult = ProcessResult(123, 1, 'Fallback failure', 'Fallback failure');
+
+    expect(
+      skiaClient.tryjobAdd('golden_file_test', goldenFile),
```

Comment:
Is there a test that tests `tryjobAdd` with `isFlaky` enabled? (or maybe there is and I'm not finding it?)

Reply:
Added! :)